insight-rest-client/src/main/java/org/nuxeo/ai/sdk/rest/api/DedupCaller.java

```diff
@@ -0,0 +1,138 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ *  Contributors:
+ *       Andrei Nechaev
+ */
+
+package org.nuxeo.ai.sdk.rest.api;
+
+import static java.util.Collections.emptyList;
+import static org.nuxeo.ai.sdk.rest.Common.DEFAULT_XPATH;
+import static org.nuxeo.ai.sdk.rest.Common.DISTANCE_PARAM;
+import static org.nuxeo.ai.sdk.rest.Common.UID;
+import static org.nuxeo.ai.sdk.rest.Common.XPATH_PARAM;
+import static org.nuxeo.ai.sdk.rest.client.InsightClient.MAPPER;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.sdk.objects.TensorInstances;
+import org.nuxeo.ai.sdk.rest.ResponseHandler;
+import org.nuxeo.ai.sdk.rest.client.API;
+import org.nuxeo.ai.sdk.rest.client.InsightClient;
+import org.nuxeo.ai.sdk.rest.exception.InvalidEndpointException;
+import org.nuxeo.ai.sdk.rest.exception.InvalidParametersException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+public class DedupCaller implements Resource {
+
+    private static final TypeReference<List<String>> LIST_TYPE_REFERENCE = new TypeReference<List<String>>() {
+    };
+
+    private final Logger log = LogManager.getLogger(DedupCaller.class);
+
+    private final InsightClient client;
+
+    private final API.Dedup type;
+
+    public DedupCaller(InsightClient client, API.Dedup type) {
+        this.client = client;
+        this.type = type;
+    }
+
+    @Nullable
+    @Override
+    public <T> T call(Map<String, Serializable> parameters) throws IOException {
+        return call(parameters, null);
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T call(Map<String, Serializable> parameters, Serializable payload) throws IOException {
+        switch (this.type) {
+        case INDEX:
+            return (T) handleIndex(parameters, (TensorInstances) payload);
+        case FIND:
+            return (T) handleFind(parameters, (TensorInstances) payload);
+        default:
+            throw new InvalidEndpointException("No such endpoint " + this.type.name());
+        }
+    }
+
+    private Boolean handleIndex(Map<String, Serializable> parameters, TensorInstances payload)
+            throws JsonProcessingException {
+        String docId = (String) parameters.get(UID);
+        String xpath = (String) parameters.getOrDefault(XPATH_PARAM, DEFAULT_XPATH);
```

Comment:
I really would prefer to always force a XPATH at this level. Define a default on ai-core if needed, but this stay open.